### DISCUSSION TITLE
[Node Operator Question] regarding L2 block sync after Ecotone

### DISCUSSION THREAD

**Discussion by changhlee** [ethereum-optimism/developers#553] [Node Operators]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I’m trying to sync L2 blocks on the Optimism mainnet and have encountered a few questions. My ultimate goal is to query historical blocks and transactions from the op-geth console for research purposes.

1. After the Ecotone upgrade, I understand that I need to configure a Blob Archiver to fetch historical blocks and transactions. Is it possible for op-node to fetch them solely through its own p2p network peers? My understanding is that op-node maintains L2 blocks on its own. As mentioned, I just want to access historical blocks and transactions on the mainnet.

2. Where can I find the genesis file for op-mainnet?

Thanks.

### Protocol Description

op-geth 1.101315.3
op-node 1.7.7
op-mainenet/ethereum

**Comment by imtipi** [comment]
> Is it possible for op-node to fetch them solely through its own p2p network peers?

in snap sync mode with full node,yes
but after you reach to the chain head(latest block),node will switch back to fetch block using blobs rpc.so you can't just p2p all the time

>  I understand that I need to configure a Blob Archiver to fetch historical blocks and transactions

it can be any third-party rpc/self-host CL client(that keeps all blobs,not deleted) that support blob_sidecars method,example:quicknode ,Blob Archiver isn't the only choice

> access historical blocks and transactions on the mainnet

you need to define how historical it is. After bedrock upgrade it is find,but if you want to access pre-bedrock historical data,it even need a old geth to fetch it(or simplified just point --rollup.historicalrpc= to https://mainnet.optimism.io/)

> Where can I find the genesis file for op-mainnet?

can I ask why you need this?

**Reply by changhlee** [reply]
Thank you for your reply.

Just to give you some context, I was running op-node and op-geth before Ecotone (fully synced), and I was able to look up transactions from block 1 to the latest block a few months ago. It's been a while since I stopped those nodes, and now I'm trying to re-sync the blocks and look up all the transactions again. I'm wondering if fetching blobs is the only way to retrieve L2 blocks older than 18 days.

Regarding the genesis file, I'm attempting to replay op-mainnet transactions in my private network built with the OP Stack. I need the genesis file to initialize the network properly; otherwise, the transactions won't run correctly. I assumed the genesis file would be publicly available, similar to Ethereum. I'm a blockchain researcher, and my questions are for research purposes.

**Reply by imtipi** [reply]
> I'm wondering if fetching blobs is the only way to retrieve L2 blocks older than 18 days

yes,and I cannot find the way to switch back snap sync in a running node,so have to use blobs

**Reply by changhlee** [reply]
I see. Thanks for your help!

**Comment by Chomtana** [comment]
> Where can I find the genesis file for op-mainnet?

https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/extra/genesis/mainnet/op.json.gz

Should be this one

**Reply by changhlee** [reply]
Thanks, but I found it odd that the op.json.gz file is unusually small and contains almost nothing. It's very different from the other genesis files for the other chains in the same folder, and even from the op.json.gz file in the Sepolia folder. Is it supposed to be correct?

**Reply by Chomtana** [reply]
OP Mainnet is distinct from other chains because it retains a legacy pre-Bedrock state. Before the implementation of snap sync, launching an OP Mainnet node required downloading a pre-made datadir, as it was not possible to perform a full sync from genesis. This suggests that either OP Mainnet does not have a traditional genesis file in this context, or it is a genesis file from the very first version of the OP chain, which dates back several years.
